### ISSUE TITLE
Can't translate a function with no arguments to c

### ISSUE THREAD

**Issue by EmilyBourne** [pyccel/pyccel#394]
**Describe the bug**
The handling of functions if the wrapper assumes that a function has arguments. As a result, we can't translate a function without arguments into c

**To Reproduce**
Provide code to reproduce the behavior:
```python
def f():
    return 1
```

The following code is generated to handle the arguments:
```c
if (!PyArg_ParseTuple(args, "", ))
        return NULL;
```
which of course leads to the following error:
```
func_wrapper.c: In function ‘f_wrapper’:
func_wrapper.c:10:37: error: expected expression before ‘)’ token
     if (!PyArg_ParseTuple(args, "", ))
                                     ^
error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
```

**Expected behavior**
An `if` condition should be added to the wrapper to only collect the arguments when they are expected

**Language**
C

**Comment by MenoIy** [comment]
If there are no arguments we have an empty list in expr. arguments we can put a condition before printing PyArgs_ParseTuple  code  like   len(arguments list ) > 0.

**Comment by EmilyBourne** [comment]
There is a test for this in `test_epyccel_functions.py`:
```
def test_func_no_args_f1():
    def f1():
        from numpy import pi
        value = (2*pi)**(3/2)
        return value

    f = epyccel(f1)
    assert abs(f()-f1()) < 1e-13
```
Once `pi` is available in C, this test (and the other non-array tests in this file) can be activated for C by modifying them as follows:
```
def test_func_no_args_f1(language):
    def f1():
        from numpy import pi
        value = (2*pi)**(3/2)
        return value

    f = epyccel(f1, language = language)
    assert abs(f()-f1()) < 1e-13
```

The remaining tests in this file can either be activated once arrays are added to C, or can be activated now but marked as expected failures. E.g:
```
def test_decorator_f2():
    @types('int [:]')
    def f2(x):
        y = x[0] - 1
        return y

    f = epyccel(f2)

    # ...
    x = np.array([3, 4, 5, 6], dtype=int)
    assert f(x) == f2(x)
    # ...

    # ...
    x = [3, 4, 5, 6]
    assert f(x) == f2(x)
    # ...
```
should become:
```
@pytest.mark.parametrize( 'lang', (
        pytest.param("fortran", marks = pytest.mark.fortran),
        pytest.param("c", marks = [
            pytest.mark.xfail(reason="Arrays are not implemented in C (see issue #312)"),
            pytest.mark.c]
        )
    )
)
def test_decorator_f2(lang):
    @types('int [:]')
    def f2(x):
        y = x[0] - 1
        return y

    f = epyccel(f2, language = lang)

    # ...
    x = np.array([3, 4, 5, 6], dtype=int)
    assert f(x) == f2(x)
    # ...

    # ...
    x = [3, 4, 5, 6]
    assert f(x) == f2(x)
    # ...
```
